app/models/user.rb

```diff
@@ -0,0 +1,22 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  property :id, Serial
+  property :email, String
+  property :name, String
+  property :username, String
+  property :password_digest, Text
+
+  validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.